diff --git a/MediaStreamContext.cpp b/MediaStreamContext.cpp
index 672d8fc..1e1d160 100644
--- a/MediaStreamContext.cpp
+++ b/MediaStreamContext.cpp
@@ -39,6 +39,8 @@ void MediaStreamContext::InjectFragmentInternal(CachedFragment* cachedFragment,
 		MediaProcessor::process_fcn_t processor = [this](AampMediaType type, SegmentInfo_t info, std::vector<uint8_t> buf)
 		{
 		};
+		AAMPLOG_DEBUG("Type[%d] cachedFragment->position: %f cachedFragment->duration: %f cachedFragment->initFragment: %d cachedFragment->PTSOffsetSec %f", type, cachedFragment->position,cachedFragment->duration,cachedFragment->initFragment,cachedFragment->PTSOffsetSec);
+
 		fragmentDiscarded = !playContext->sendSegment( &cachedFragment->fragment, cachedFragment->position,
 														cachedFragment->duration, cachedFragment->PTSOffsetSec, isDiscontinuity, cachedFragment->initFragment, processor, ptsError);
 	}
@@ -366,6 +368,7 @@ bool MediaStreamContext::CacheFragment(std::string fragmentUrl, unsigned int cur
 	else
 	{
 		cachedFragment->position = position;
+		AAMPLOG_INFO("[pto]Type[%d] fragmentUrl %s position %f duration %f discontinuity %d pto %f scale %u", type, fragmentUrl.c_str(), position, fragmentDurationS, discontinuity, pto, scale);
 		cachedFragment->duration = fragmentDurationS;
 		cachedFragment->discontinuity = discontinuity;
 		segDLFailCount = 0;
diff --git a/fragmentcollector_mpd.cpp b/fragmentcollector_mpd.cpp
index ee569a4..61612e5 100644
--- a/fragmentcollector_mpd.cpp
+++ b/fragmentcollector_mpd.cpp
@@ -159,8 +159,7 @@ StreamAbstractionAAMP_MPD::StreamAbstractionAAMP_MPD(class PrivateInstanceAAMP *
 	,mAudioSurplus(0)
 	,mVideoSurplus(0)
 	,mIsSegmentTimelineEnabled(false)
-	,mSeekedInPeriod(false)
-	,mIsFinalFirstPTS(false)
+	,mSeekedInPeriod(false), mIsFinalFirstPTS(false)
 {
 	this->aamp = aamp;
 	if (aamp->mDRMLicenseManager)
@@ -893,6 +892,11 @@ bool StreamAbstractionAAMP_MPD::FetchFragment(MediaStreamContext *pMediaStreamCo
 			pMediaStreamContext->initialization = std::string(fragmentUrl);
 		}
 	}
+
+	AAMPLOG_INFO("[Fetch] %s position %f fragmentDuration %f type %s",
+					fragmentUrl.c_str(), position, fragmentDuration,GetMediaTypeName(pMediaStreamContext->mediaType));
+
+
 	bool fragmentCached = pMediaStreamContext->CacheFragment(fragmentUrl, curlInstance, position, fragmentDuration, NULL, isInitializationSegment, discontinuity
 		,(mCdaiObject->mAdState == AdState::IN_ADBREAK_AD_PLAYING), pto, timeScale);
 	// Check if we have downloaded the fragment and waiting for init fragment download on
@@ -2037,6 +2041,9 @@ bool StreamAbstractionAAMP_MPD::PushNextFragment( class MediaStreamContext *pMed
 						snprintf(nextrange, sizeof(nextrange), "%" PRIu64 "-%" PRIu64 "",nextfragmentOffset, nextfragmentOffset+nextReferencedSize - 1);
 						setNextRangeRequest(fragmentUrl,nextrange,(&pMediaStreamContext->fragmentDescriptor)->Bandwidth,AampMediaType(pMediaStreamContext->type));
 					}
+					AAMPLOG_INFO("[Fetch] %s fragmentTime %f fragmentDuration %f type %s",
+					fragmentUrl.c_str(), pMediaStreamContext->fragmentTime, fragmentDuration,GetMediaTypeName(pMediaStreamContext->mediaType));
+
 					if(pMediaStreamContext->CacheFragment(fragmentUrl, curlInstance, pMediaStreamContext->fragmentTime, fragmentDuration, range ))
 					{
 						pMediaStreamContext->fragmentTime += fragmentDuration;
@@ -2099,6 +2106,10 @@ bool StreamAbstractionAAMP_MPD::PushNextFragment( class MediaStreamContext *pMed
 								setNextobjectrequestUrl(nextsegmentURL->GetMediaURI(),&pMediaStreamContext->fragmentDescriptor,AampMediaType(pMediaStreamContext->type));
 							}
 							double fragmentDurationS = ComputeFragmentDuration(segmentList->GetDuration(), segmentList->GetTimescale());
+								AAMPLOG_INFO("[Fetch] %s fragmentTime %f type %s",
+					fragmentUrl.c_str(), pMediaStreamContext->fragmentTime,GetMediaTypeName(pMediaStreamContext->mediaType));
+
+							
 							if( pMediaStreamContext->CacheFragment(fragmentUrl, curlInstance, pMediaStreamContext->fragmentTime, fragmentDurationS, segmentURL->GetMediaRange().c_str() ) )
 							{
 								pMediaStreamContext->fragmentTime += fragmentDurationS;
@@ -2429,8 +2440,8 @@ double StreamAbstractionAAMP_MPD::SkipFragments( MediaStreamContext *pMediaStrea
 			if (pto > startTime)
 			{
 				double offset = (double)(pto - startTime) / (double)segmentTemplates.GetTimescale();
-				AAMPLOG_INFO("Adding PTO offset:%lf to skipTime: %lf", offset, skipTime);
 				skipTime += offset;
+				AAMPLOG_INFO("Adding PTO offset:%lf to skipTime: %lf", offset, skipTime);
 			}
 		}
 		do
@@ -2487,6 +2498,7 @@ double StreamAbstractionAAMP_MPD::SkipFragments( MediaStreamContext *pMediaStrea
 					double fragmentDuration = ComputeFragmentDuration(duration,timeScale);
 					double nextPTS = (double)(pMediaStreamContext->fragmentDescriptor.Time + duration)/timeScale;
 					double firstPTS = (double)pMediaStreamContext->fragmentDescriptor.Time/timeScale;
+//					AAMPLOG_TRACE("[%s] firstPTS %f nextPTS %f duration %f skipTime %f", pMediaStreamContext->name, firstPTS, nextPTS, fragmentDuration, skipTime);
 					if (firstFrag && updateFirstPTS)
 					{
 						if (skipToEnd)
@@ -2511,6 +2523,7 @@ double StreamAbstractionAAMP_MPD::SkipFragments( MediaStreamContext *pMediaStrea
 								double newSkipTime = skipTime + (mFirstVideoFragPTS - firstPTS); /* adjust for audio/video frag start PTS differences */
 								newSkipTime -= mVideoPosRemainder;   /* adjust for mVideoPosRemainder, which is (video seekposition/skipTime - mFirstPTS) */
 								newSkipTime += fragmentDuration/4.0; /* adjust for case where video start is near end of current audio fragment by adding to the audio skipTime, pushing it into the next fragment, if close(within this adjustment) */
+								//AAMPLOG_INFO("newSkiptime %f, skipTime %f  mFirstFragPTS[vid] %f  firstPTS %f  mFirstFragPTS[vid]-firstPTS %f mVideoPosRemainder %f  fragmentDuration/4.0 %f", newSkipTime, skipTime, mFirstVideoFragPTS, firstPTS, mFirstVideoFragPTS-firstPTS, mVideoPosRemainder,  fragmentDuration/4.0);
 								skipTime = newSkipTime;
 							}
 						}
@@ -2756,7 +2769,7 @@ double StreamAbstractionAAMP_MPD::SkipFragments( MediaStreamContext *pMediaStrea
 				}
 			}
 			if( skipTime==0 ) AAMPLOG_WARN( "skipTime is 0" );
-		} while(true);
+		} while(true); // was while(skipTime != 0);
 
 		AAMPLOG_INFO("Exit :Type[%d] timeLineIndex %d fragmentRepeatCount %d fragmentDescriptor.Number %" PRIu64 " fragmentTime %f FTime:%f",pMediaStreamContext->type,
 				pMediaStreamContext->timeLineIndex, pMediaStreamContext->fragmentRepeatCount, pMediaStreamContext->fragmentDescriptor.Number, pMediaStreamContext->fragmentTime,pMediaStreamContext->fragmentDescriptor.Time);
@@ -8640,6 +8653,9 @@ void StreamAbstractionAAMP_MPD::FetchAndInjectInitialization(int trackIdx, bool
 							{
 								setNextRangeRequest(fragmentUrl,nextrange,(&pMediaStreamContext->fragmentDescriptor)->Bandwidth,AampMediaType(pMediaStreamContext->type));
 							}
+								AAMPLOG_INFO("[Fetch] %s fragmentTime %f type %s",
+					fragmentUrl.c_str(), pMediaStreamContext->fragmentTime,GetMediaTypeName(pMediaStreamContext->mediaType));
+
 							if(!pMediaStreamContext->CacheFragment(fragmentUrl,
 								getCurlInstanceByMediaType(pMediaStreamContext->mediaType),
 								pMediaStreamContext->fragmentTime,
@@ -8738,6 +8754,9 @@ void StreamAbstractionAAMP_MPD::FetchAndInjectInitialization(int trackIdx, bool
 										{
 											setNextobjectrequestUrl(nextsegurl->GetMediaURI(),&pMediaStreamContext->fragmentDescriptor,AampMediaType(pMediaStreamContext->type));
 										}
+											AAMPLOG_INFO("[Fetch] %s fragmentTime %f type %s",
+					fragmentUrl.c_str(), pMediaStreamContext->fragmentTime,GetMediaTypeName(pMediaStreamContext->mediaType));
+
 										if(!pMediaStreamContext->CacheFragment(fragmentUrl,
 												getCurlInstanceByMediaType(pMediaStreamContext->mediaType),
 												pMediaStreamContext->fragmentTime,
@@ -8844,6 +8863,9 @@ void StreamAbstractionAAMP_MPD::CacheEncryptedHeader(int trackIdx, std::string h
 		bool temp = false;
 		try
 		{
+				AAMPLOG_INFO("[Fetch] %s fragmentTime %f ",
+					headerUrl.c_str(), mMediaStreamContext[trackIdx]->fragmentTime);
+
 			temp =  mMediaStreamContext[trackIdx]->CacheFragment(headerUrl, (eCURLINSTANCE_VIDEO + mMediaStreamContext[trackIdx]->mediaType), mMediaStreamContext[trackIdx]->fragmentTime, 0.0, NULL, true, false, false, 0, 0, true);
 		}
 		catch(const std::regex_error& e)
diff --git a/priv_aamp.cpp b/priv_aamp.cpp
index 6bb169c..831b001 100644
--- a/priv_aamp.cpp
+++ b/priv_aamp.cpp
@@ -5116,7 +5116,7 @@ void PrivateInstanceAAMP::TuneHelper(TuneType tuneType, bool seekWhilePaused)
 		//Only trigger the clear to encrypted pipeline switch while on retune
 		mEncryptedPeriodFound = false;
 		mPipelineIsClear = false;
-		AAMPLOG_INFO ("Resetting mClearPipeline & mEncryptedPeriodFound");
+		AAMPLOG_INFO ("[Lashmintha_1]Resetting mClearPipeline & mEncryptedPeriodFound");
 	}
 
 	TeardownStream(newTune|| (eTUNETYPE_RETUNE == tuneType));
diff --git a/streamabstraction.cpp b/streamabstraction.cpp
index 087b511..7b7282f 100644
--- a/streamabstraction.cpp
+++ b/streamabstraction.cpp
@@ -1355,7 +1355,10 @@ void MediaTrack::ProcessAndInjectFragment(CachedFragment *cachedFragment, bool f
 				{
 					// We could skip RestampPts when PTSOffsetSec==0 but the RestampPts log line
 					// would then be missing and it is important for l2 tests
+
 					int64_t ptsOffset = cachedFragment->PTSOffsetSec * cachedFragment->timeScale;
+
+					AAMPLOG_INFO("[pto]%s Restamping PTS for fragment %s with offset %" PRIu64 "", name, cachedFragment->uri.c_str(), ptsOffset);
 					(void)mIsoBmffHelper->RestampPts(cachedFragment->fragment, ptsOffset,
 													 cachedFragment->uri, name,
 													 cachedFragment->timeScale);
@@ -1413,10 +1416,13 @@ void MediaTrack::ProcessAndInjectFragment(CachedFragment *cachedFragment, bool f
 		{
 			if(AAMP_NORMAL_PLAY_RATE==aamp->rate)
 			{
+				AAMPLOG_INFO("[pto] Injecting fragment for %s pos=%f dur=%f discontinuity=%d", name,cachedFragment->position, cachedFragment->duration, cachedFragment->discontinuity);
 				InjectFragmentInternal(cachedFragment, fragmentDiscarded, isDiscontinuity);
 			}
 			else
 			{
+				AAMPLOG_INFO("[pto] Injecting fragment for %s pos=%f dur=%f discontinuity=%d", name,cachedFragment->position, cachedFragment->duration, cachedFragment->discontinuity);
+
 				InjectFragmentInternal(cachedFragment, fragmentDiscarded, cachedFragment->discontinuity);
 			}
 		}
@@ -1425,7 +1431,7 @@ void MediaTrack::ProcessAndInjectFragment(CachedFragment *cachedFragment, bool f
 		{
 			pContext->NotifyBitRateUpdate(cachedFragment->profileIndex, cachedFragment->cacheFragStreamInfo, cachedFragment->position);
 		}
-		AAMPLOG_DEBUG("%s - injected cached fragment at pos %f dur %f", name, cachedFragment->position, cachedFragment->duration);
+		AAMPLOG_INFO("[pto]%s - injected cached fragment at pos %f dur %f", name, cachedFragment->position, cachedFragment->duration);
 		if (!fragmentDiscarded)
 		{
 			std::lock_guard<std::mutex> lock(mTrackParamsMutex);
