diff --git a/src/ipp2/CMakeLists.txt b/src/ipp2/CMakeLists.txt
index 9931cc0..ca5df41 100644
--- a/src/ipp2/CMakeLists.txt
+++ b/src/ipp2/CMakeLists.txt
@@ -174,6 +174,8 @@ if(ENABLE_DS_CLIENT)
 endif(ENABLE_DS_CLIENT)
 include_directories(${Boost_INCLUDE_DIRS})
 include_directories(${JSONCPP_INCLUDE_DIRS})
+include_directories(${LIBWEBSOCKETPP_INCLUDE_DIRS})
+include_directories(${TINYXML2_INCLUDE_DIR})
 
 find_path(ASIO_PATH asio.hpp)
 if(ASIO_PATH)
